src/libs/API/parameters/UpdatePolicyConnectionConfigurationParams.ts

```diff
@@ -0,0 +1,10 @@
+import type {ConnectionName} from '@src/types/onyx/Policy';
+
+type UpdatePolicyConnectionConfigurationParams = {
+    policyID: string;
+    connectionName: ConnectionName;
+    settingName: string;
+    settingValue: unknown;
+};
```

Comment:
I think this type can be made tighter:

```suggestion
type UpdatePolicyConnectionConfigurationParams<T extends ConnectionName, S extends keyof Connections[T]['config']> = {
    policyID: string;
    connectionName: T;
    settingName: S;
    settingValue: Connections[T]['config'][S];
};
```

Then the `settingValue` is really tied to what you put in `settingName` and also in `connectionName`

For example, typescript is fine with this:

<img width="717" alt="image" src="https://github.com/Expensify/App/assets/87341702/4221c263-5d5d-46dc-964f-b91e096efedb">


but it doesn't like 

<img width="684" alt="image" src="https://github.com/Expensify/App/assets/87341702/c2c93c2c-adaf-4d46-8229-904e72c32f2d">


Notice the typo in jobId

Reply:
https://github.com/Expensify/App/pull/38006/commits/fb2f4f7d4aa2a362bbfca38581e2c87410c985f0